src/main/java/com/osiris/autoplug/client/tasks/updater/server/TaskServerUpdater.java

```diff
@@ -403,114 +300,31 @@ private void doFabricUpdatingLogic() throws WrongJsonTypeException, IOException,
-                            if (final_dest == null)
-                                final_dest = new File(GD.WORKING_DIR + "/" + serverSoftware + "-latest.jar");
-                            if (final_dest.exists()) final_dest.delete();
-                            final_dest.createNewFile();
-                            FileUtils.copyFile(cache_dest, final_dest);
-                            setStatus("Server update was installed successfully (" + build_id + " -> " + latest_build_id + ")!");
-                            updaterConfig.server_build_id.setValues("" + latest_build_id);
-                            updaterConfig.save();
-                            setSuccess(true);
-                        } else {
-                            if (updaterConfig.server_skip_hash_check.asBoolean()) {
-                                addWarning("Note that the hash check failed for this download, but installing anyways because skip-hash-check is enabled!");
-                                File final_dest = serverExe;
-                                if (final_dest == null)
-                                    final_dest = new File(GD.WORKING_DIR + "/" + serverSoftware + "-latest.jar");
-                                if (final_dest.exists()) final_dest.delete();
-                                final_dest.createNewFile();
-                                FileUtils.copyFile(cache_dest, final_dest);
-                                setStatus("Server update was installed successfully (" + build_id + " -> " + latest_build_id + ")!");
-                                updaterConfig.server_build_id.setValues("" + latest_build_id);
-                                updaterConfig.save();
-                                setSuccess(true);
-                            } else {
-                                setStatus("Downloaded server update is broken (hash check failed). Nothing changed!");
-                                setSuccess(false);
-                            }
-                        }
-
-                    } else {
-                        setStatus("Server update failed!");
-                        setSuccess(false);
-                    }
-                    break;
-                }
+            setStatus("Starting MCServer updater...");
+            File outputFile;
+            if (profile.equals("MANUAL")) {
+                outputFile = new File(downloadsDir.getAbsolutePath() + "/" + serverSoftware + "-latest.jar");
+            } else if (serverExe == null) {
+                outputFile = new File(GD.WORKING_DIR + "/" + serverSoftware + "-latest.jar");
+            } else {
+                outputFile = serverExe;
+            }
+            try {
+                UpdateStatus status = UpdateBuilder.updateProject(serverSoftware)
+                        .version(serverVersion)
+                        .outputFile(outputFile)
+                        .execute();
+                setStatus(status.getMessage());
```

Comment:
What are the possible messages here? Also is the hash checked?

I would like to keep the old message format: `setStatus("Server update was installed successfully (" + build_id + " -> " + latest_build_id + ")!");` so that there isn't a major change in format between different updaters (plugins & mods updaters for example use a similar formatting).

Reply:
https://github.com/HSGamer/MCServerUpdater/blob/master/mc-server-updater-lib/src/main/java/me/hsgamer/mcserverupdater/UpdateStatus.java

Reply:
build_id should not be used because some versions only need checking the file hash, others store the "checksum" as a file.

Reply:
the "checksum" of MCServerUpdater is now stored in `build_id`. Check again.